doc/sphinx/source/fota.txt

```diff
@@ -0,0 +1,27 @@
+######################
+FOTA software of MuxPi
+######################
+
+This manual presents the main functionalities of fota.
+
+The MuxPi together with NanoPi Neo have two microSD card slots. The slot on the NanoPi Neo is used to boot up the MuxPi and install software such as fota or stm. When using FOTA, the card slot on the muxPi must be used.
+
+Find the correct device by running ``lsblk``
+
+**********
+Fota usage
+**********
+
+Fota requires two arguments. The device to flash (e.g /dev/sda) and a json file.
```

Comment:
It will be also good to describe md5 parameter.
Anyway even with your json map example I am still not able to get fota work properly.

root@nanopineo:~# ls -la boot.img 
-rw-r--r-- 1 root root 80915 Feb 22 06:57 boot.img
root@nanopineo:~# ls -la /root/boot.img 
-rw-r--r-- 1 root root 80915 Feb 22 06:57 /root/boot.img
root@nanopineo:~# ls -la /dev/sda*
brw-rw---- 1 root disk 8, 0 Feb 22 06:47 /dev/sda
brw-rw---- 1 root disk 8, 1 Feb 22 06:47 /dev/sda1
brw-rw---- 1 root disk 8, 2 Feb 22 06:47 /dev/sda2
root@nanopineo:~# cat map.json 
{"/root/boot.img": "1", "boot.img": "2"}
root@nanopineo:~# fota -card /dev/sda -map map.json
2019/02/22 06:57:52 FOTA initialized
2019/02/22 06:57:52 SDcard detected
2019/02/22 06:57:52 nothing to do

Reply:
Yeah, I didn't notice it at first, but the docs are a bit incorrect. You have to provide one (or more) paths or URLs leading to .tar.gz files and the keys in the json have to be names of the files in the .tar.gz files. So if you have a boot.tar.gz with boot.img and modules.img and userland.tar.gz with rootfs.img you have to make a map.json file with something like `{"boot.img": "1", "rootfs.img": "2", "modules.img": "3"}` and run `fota -card /dev/sda -map map.json boot.tar.gz userland.tar.gz`.

Reply:
How is that logic around md5sum? Is it md5sum of that tar.gz files?

Reply:
Sorry, I don't remember. I never used that feature.

Reply:
ok. It is md5sum for tar.gz files. It will be also good to say that it is doing in general dd if=boot.img of=/dev/sdaX and it is not copying files to file system on that partition. 
Or is there also a support for that?

Reply:
I don't recall any support for that indeed. Fota is just good at flashing Tizen nightly images.

Reply:
Whats the purpose of the md5sum if its for the tar.gz files? wouldnt it be better to be for the image instead? In that case you would also be able to verify if you've flashed correctly. Is this possible with a tarball md5sum?

Reply:
I added a line where it says that it only dd-s the image.

Reply:
> I don't recall any support for that indeed. Fota is just good at flashing Tizen nightly images.

This is a MUXPI thread but I am curious how to use this functionality in connection to the whole Samsung stack. Will be good to get that examples in that repos.